viewing_party/party.py

```diff
@@ -0,0 +1,253 @@
+
+    fav_genres = {}
+    if len(watched) == 0:
+        return None 
+    else:
+        for genre in genres:
+            if genre not in fav_genres:
+                fav_genres[genre] = 1
+            else:
+                fav_genres[genre] += 1
+    
+    favorite_genre = max(fav_genres, key=fav_genres.get)
+    
+    return favorite_genre
+
+
+# Wave 3 **********************************************
+
+# 1
+
+def get_unique_watched(user_data):
+    unique_watched_list = []
+    for dicts in user_data["watched"]:
+        unique_watched_list.append(dicts)
+    
+    friends_watched_list = []
+    for friends_watched in user_data["friends"]:
+        for friends_dicts in friends_watched["watched"]:
+            friends_watched_list.append(friends_dicts)
+
+    unique_watched_set = set()
+    friends_watched_set = set()
+
+    if len(user_data["watched"]) == 0:
+        return []
+    else:
+        for movie_dict in unique_watched_list:
+            for movie in movie_dict.values():
+                unique_watched_set.add(movie)
+        for friends_movie_dicts in friends_watched_list:
+            for title in friends_movie_dicts.values():
+                friends_watched_set.add(title)
+
+    set_difference = unique_watched_set - friends_watched_set
+    unique_watched_list = list(set_difference)
+
+    users_unique_watched_list = []
+    for unique_movies in unique_watched_list:
+        list_of_dicts = {"title": unique_movies}
+        users_unique_watched_list.append(list_of_dicts)
+
+    return users_unique_watched_list
+
+
+# 2
+
+def get_friends_unique_watched(user_data):
+    user_watched_list = []
+    for user_dicts in user_data["watched"]:
+        user_watched_list.append(user_dicts)
+
+    friends_watched_list = []
+    for friend_list in user_data["friends"]:
+        for friend_dicts in friend_list["watched"]:
+            friends_watched_list.append(friend_dicts)
+
+    user_watched_set = set()
+    friends_watched_set = set()
+    for list_of_dicts in user_data["friends"]:
+        for watched_list in list_of_dicts["watched"]:
+            if len(watched_list) == 0:
+                return []
+        else: 
+            for movie_dict in user_watched_list:
+                for movie in movie_dict.values():
+                    user_watched_set.add(movie)
+            for friends_movie_dicts in friends_watched_list:
+                for title in friends_movie_dicts.values():
+                    friends_watched_set.add(title)
+
+    set_difference = friends_watched_set - user_watched_set
+    unique_watched_list = list(set_difference)
+
+    friends_unique_watched_list = []
+    for unique_movies in unique_watched_list:
+        list_of_dicts = {"title": unique_movies}
+        friends_unique_watched_list.append(list_of_dicts)
+
+    return friends_unique_watched_list
+
+
+# Wave 4 **********************************************
+
+# 1
+
+def get_available_recs(user_data):
+    user_movies = []
```

Comment:
Consider using a set here to improve the `in` lookup on line 184 from O(n) to O(1).